### ISSUE TITLE
PHP Warning:  simplexml_load_string(): Entity: line 1: parser error

### ISSUE THREAD

**Issue by tgd1973** [DanielnetoDotCom/YouPHPTube#338]
An error that if I leave the Live plugin enabled without having loaded NGINX on the server ... it fills the log youphptube.log to the  1 -5mb values.  Apart from disabling the LIVE plug ... What should I do?
Then another thing since we are at Live ... I have not seen any reference to the Recording plug that sends the file to encoder. Kindly can you make a tuttorial of how it works?

[29-Jan-2018 10:35:39 Europe/Berlin] PHP Warning:  simplexml_load_string(): Entity: line 1: parser error : SYSTEM or PUBLIC, the URI is missing in /var/www/clients/client1/web6/web/plugin/Live/Live.php on line 81
[29-Jan-2018 10:35:39 Europe/Berlin] PHP Warning:  simplexml_load_string(): &lt;!DOCTYPE HTML PUBLIC &quot;-//IETF//DTD HTML 2.0//EN&quot;&gt; in /var/www/clients/client1/web6/web/plugin/Live/Live.php on line 81
[29-Jan-2018 10:35:39 Europe/Berlin] PHP Warning:  simplexml_load_string():    

Then another problem related to the NGINX server ... in on a server with Live option,  always active ... if a person will be able to find out  this rtmp server,  in theory it could exploit it ... for its own purposes. ... To be sure that nobody uses our band, it must be create a kind of firewall that leaves in input, for NGINX-RTMP only the client-encoder with the key that is generated in the control panel of the LIVE ... I know it's complicated to do, but almost all video streaming servers. have this option to input rtmp with an incoming password (Wowza, Flussonic ...)

If I want to use my Flussonic server instead of NGINX for live ... how should I proceed?

**Comment by DanielnetoDotCom** [comment]
Hi, to auto record live stream check this https://github.com/DanielnetoDotCom/YouPHPTube/wiki/Record-Live-Stream

About the stream protection, it is secure, you need the stream key to be able to connect on the NGINX, only the stream owner should have it
![image](https://user-images.githubusercontent.com/18009293/35507248-e3efbac2-04ca-11e8-822f-13fd714b9985.png)

About the Flussonic I never use it before

**Comment by tgd1973** [comment]
Ok if you send the signal to  ...com:444/live/?P = .... if not the right one does not accept it but if I send it directly to   ....com:444   how do you avoid it? 
Then this .flv I use it in one of my embedded player .flv I'm sure it goes ... 
It's also true that no on use anymore flash players  ... having almost completely passed to HTML5

Another question ...  with this plug recording how many qualities do you get from one video? Will it always be 3 qualities? (as also from Youtube) I wonder, because I do not see a multi-quality implementation in Live streaming ... in theory you have to use a transcoder that from one streaming transmitted with high quality, with my encoder,  the server also get the other lower quality ...

**Comment by DanielnetoDotCom** [comment]
com:444 is only a welcome page of nginx

you can configure the live stream quality that you want direct into nginx.
But when it is record it will be the qualities you configure on youphptube (max 3)

**Comment by tgd1973** [comment]
I did some tests and it seems safe ... Congratulations ... you managed to close some unauthorized entries ... that instead other sites have ...

I do not understand, however, having bought the "VideoResolutionSwitcher" plug  can change the resolution ... with the "SendRecordedToEncoder" plug for the video sent to the encoder? Will  "Low, SD, HS" encoder options still be active? and how long can I record?  (for example if I have 750mb the size of the files for encoder ...)

**Comment by DanielnetoDotCom** [comment]
How long you can record it depends your server configuration: max file upload size and drive free space.

about the resolution you want to select to encode check your plugins menu
![image](https://user-images.githubusercontent.com/18009293/35511638-70a4d136-04db-11e8-9aba-74fbeef52b89.png)

**Comment by tgd1973** [comment]
It would be very interesting if the NGINX server could also be installed in another PC or better to have more NGINX servers on which to allocate the load ... and be able to manage users in ways that do not overload a single live server.     
For example if one wants to manage more churches with the site youphptube ... A live from a one church to go on a server, another live on another nginx server ...

**Comment by DanielnetoDotCom** [comment]
Yeah, agree, I am planning to add support to play videos from external streamers, like wowza and others

**Comment by tgd1973** [comment]
When you want to make test for a module to support other servers ... also consider the Flussonic as a mediaserver.

**Comment by tgd1973** [comment]
Back with the problems ... for 2 hours I'm trying to solve these three variables ... shared server Ubuntu with ISPconfig + nat (public IP on the router wan 192.168.xx on the LAN then the IP resolves on the DNS) + Https  ...  practically I arrive at this passage and I stop here:
"and edit the lines:
on_publish http://[YouPHPTubeURL]/plugin/Live/on_publish.php;
on_play http://[YouPHPTubeURL]/plugin/Live/on_play.php;
on_record_done http://[YouPHPTubeURL]/plugin/Live/on_record_done.php;

what ip I put? (I have two IP, external and internal then also the domain)  
https://tutorials.youphptube.com/video/10-min-youphptube-stream-server-installation1
your movie is all done on a local server ... without domain and without being in shared ... then the problem of reverse proxy on Apache to have https is not very inspiring because it loads so much the server (once nginx and then also apache) and anyway how do you go on a shared server?

**Comment by DanielnetoDotCom** [comment]
port 444 is nginx

your nginx mus be able to access the address: `on_publish http://[YouPHPTubeURL]/plugin/Live/on_publish.php;`
so put the ip that nginx can use to access the streamer

**Comment by tgd1973** [comment]
as in the image above ... allows publish but in a different way ... you see in the image above ... then in the page xxx/plugin/Live/?u=abc   of the youphptube does not start the play because it does not manage to download the file index.m3u8

**Comment by DanielnetoDotCom** [comment]
if you use https in your streamer, your nginx must be also https.

**Comment by tgd1973** [comment]
how do you get nginx to answer on HTTPS?

**Comment by DanielnetoDotCom** [comment]
try lets encrypt

**Comment by tgd1973** [comment]
but nginx does not answer in any way ...

**Comment by DanielnetoDotCom** [comment]
??? what do you mean? this is a nginx on https https://live.youphptube.com:444/

**Comment by tgd1973** [comment]
I discovered that port 444 is occupied with another apache service ... so I can not use it ...  nginx on the https is not responding ...

**Comment by DanielnetoDotCom** [comment]
you can use any port for https, try 445 or 446

**Comment by tgd1973** [comment]
I managed to do the proxy bypass via apache by adding to a subdomain of the domain where the youphptube was installed, the following code inside the ISPConfig > Sites > domain.tld > option >Apache directives> this code:  SSLProxyEngine On
SSLProxyVerify none
SSLProxyCheckPeerCN off
SSLProxyCheckPeerName off
SSLProxyCheckPeerExpire off
ProxyPass / https://127.0.0.1:445/
ProxyPassReverse / https://127.0.0.1:445/

then command:
 sudo a2enmod proxy
 service apache2 restart

**Comment by tgd1973** [comment]
I found the problem ... I do not know how to do it but my nginx server does not give the stream number in the statistics page ...

![b](https://user-images.githubusercontent.com/15127860/35539759-378351de-0553-11e8-9f35-896f281ecd1c.png)

 as your server does ...

![a](https://user-images.githubusercontent.com/15127860/35539769-3ff725de-0553-11e8-8ff6-c56df14689dc.png)

what do I have to do to fix it?

**Comment by DanielnetoDotCom** [comment]
What are you using to stream videos?
What parameters are you put on it :)

**Comment by tgd1973** [comment]
To make the stream use wirecast as rtmp streaming ...
the parameters are the ones that supply me the YouPhPTube interface ... in the LIVE menu:
rtmp://abc.eu/live?p=d4a658c9591a46556f83a4ad1f5d3430
and the key ...

**Comment by DanielnetoDotCom** [comment]
Hi,

The only problem I can see is the Live Stream name, it suppose to be the stream key, not a full URL

**Comment by tgd1973** [comment]
yes I think the same ... but also inserting in the wirecast the key ... in the panel of statistics does not appear ... why? something prevents ... What is the mechanism with which the Live plug works? I also saw the port god in the Live.php file that can be changed ...

**Comment by DanielnetoDotCom** [comment]
The live plugin and the users live plugin uses the stream key to identify who is transmiting

**Comment by tgd1973** [comment]
I used port 445 ... do I have to set these in Live.php too? Then I used the Proxy in apache to have https letsencrypt to be able to transmit the index.m3u8 file on secure connection ...

**Comment by DanielnetoDotCom** [comment]
everything you need to set up is on the plugin menu, you should not touch the file

**Comment by tgd1973** [comment]
I refer to this:

 public function getEmptyDataObject() {
        global $global;
        $server = parse_url($global['webSiteRootURL']);
        $obj = new stdClass();
        $obj->button_title = "LIVE";
        $obj->server = "rtmp://{$server['host']}/live";
        $obj->playerServer = "http://{$server['host']}:8080/live";
        // for secure connections
        //$obj->playerServer = "https://{$server['host']}:444/live";
        $obj->stats = "http://{$server['host']}:8080/stat";
        $obj->disableGifThumbs = false;
        return $obj;

**Comment by DanielnetoDotCom** [comment]
You should change it in the plugin

![image](https://user-images.githubusercontent.com/18009293/35571305-77a90436-05b0-11e8-9dfd-6ac04070c344.png)

**Comment by tgd1973** [comment]
OK I changed it too but it does not work ... 
There's something that prevents the signal recognition code key from being acknowledged that wirecast encoder sends to the Nginx-rtmp ... what should I do?

**Comment by DanielnetoDotCom** [comment]
I do not think the problem is wirecast, but FYI I always test with OBS, you may try to use this to see if anything changes

**Comment by tgd1973** [comment]
It does not change anything ... so it's something in the Nginx-rtmp settings because it can not get the codes from the youphpTube software ... to use them as transmit ID ...

**Comment by tgd1973** [comment]
Hi! I still could not solve the Ofline problem in Live streaming which I believe is due to a problem between Nginx and youphpTube. In time I discovered another problem ... For the uploaded videos, the display counter is no longer incremented ... I tried to re-load the whole YouphpTube script ... but the problem persists ... it can get me a point of where do I start? How does the counter work and could it be the reason for not working?
Thanks again
